kbase-extension/static/kbase/js/common/cellComponents/tabs/jobStatus/jobStatusTable.js

```diff
@@ -324,44 +337,65 @@ define([
                     jobArray.forEach((jobState) => updateJobStatusInTable(jobState));
                 },
             });
+
+            const batchId = jobManager.model.getItem('exec.jobState.job_id');
             const paramsRequired = [];
             const jobIdList = [];
             jobs.forEach((jobState) => {
-                jobIdList.push(jobState.job_id);
-                if (!jobManager.model.getItem(`exec.jobs.params.${jobState.job_id}`)) {
-                    paramsRequired.push(jobState.job_id);
+                if (!jobState.batch_job) {
+                    jobIdList.push(jobState.job_id);
+                    if (
+                        !jobManager.model.getItem(`exec.jobs.params.${jobState.job_id}`) &&
+                        jobState.job_id !== batchId
+                    ) {
+                        paramsRequired.push(jobState.job_id);
+                    }
                 }
             });
+            jobManager.addListener('job-status', [batchId].concat(jobIdList));
 
             if (paramsRequired.length) {
                 jobManager.addListener('job-info', paramsRequired, {
                     jobStatusTable_info: handleJobInfo,
                 });
-                bus.emit('request-job-info', {
-                    jobIdList: paramsRequired,
-                });
+                const jobInfoRequestParams =
+                    paramsRequired.length === jobIdList.length
+                        ? { batchId }
+                        : { jobIdList: paramsRequired };
+                bus.emit('request-job-info', jobInfoRequestParams);
```

Comment:
Request the params for whichever jobs don't yet have them. If all jobs are missing params, use the handy new `job_info_batch` method in the narrative backend.

Reply:
Why not just use `job_info_batch` anyway? I forget if it only takes a batch job id, or if it can take an arbitrary list of job ids.

I guess, regardless, it gets passed through `request-job-info`, so that's moot.